src/cdomains/intDomain.ml

```diff
@@ -919,22 +932,535 @@ struct
+    two_interval_sets_to_events x y |> 
+    combined_event_list `Join |>
+    events_to_intervals |>
+    remove_empty_gaps
+
+  let meet ik (x: t) (y: t): t = 
+    two_interval_sets_to_events x y |> 
+    combined_event_list  `Meet |> 
+    events_to_intervals |> 
+    remove_empty_gaps
+
+  let to_int = function [(x, y)] when Ints_t.compare x y = 0 -> Some x | _ -> None
+
+  let zero = [(Ints_t.zero, Ints_t.zero)]
+  let one = [(Ints_t.one, Ints_t.one)]
+
+  let top_bool = [(Ints_t.zero, Ints_t.one)]
+
+  let not_bool (x:t) = 
+    let is_false = fun x -> x = zero in
+    let is_true =  fun x -> x = one in
+    if is_true x then zero else if is_false x then one else top_bool 
+
+  let to_bool = function  
+    | [(l,u)] when Ints_t.compare l Ints_t.zero = 0 && Ints_t.compare u Ints_t.zero = 0 -> Some false
+    | x -> if leq zero x then None else Some true
+
+  let of_bool _ = function true -> one | false -> zero 
+
+  let of_interval ik (x, y) = norm_interval ik @@ Some (x, y)
+
+  let of_int ik (x: int_t) = of_interval ik (x, x)
+  let lt ik x y = 
+    match x, y with 
+    | [], [] -> bot_of ik
+    | [], _ | _, [] -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
+    | _, _ ->
+      let (max_x, min_y) = (maximal x |> Option.get , minimal y |> Option.get) in
+      let (min_x, max_y) = (minimal x |> Option.get , maximal y |> Option.get) in
+      if max_x < min_y then 
+        of_bool ik true
+      else
+      if min_x >= max_y then of_bool ik false else top_bool
+
+  let le ik x y =
+    match x, y with 
+    | [], [] -> bot_of ik
+    | [], _ | _, [] -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
+    | _, _ ->
+      let (max_x, min_y) = (maximal x |> Option.get , minimal y |> Option.get) in
+      let (min_x, max_y) = (minimal x |> Option.get , maximal y |> Option.get) in
+      if max_x <= min_y then 
+        of_bool ik true
+      else
+      if min_x > max_y then of_bool ik false else top_bool
+
+  let gt ik x y = not_bool @@ le ik x y
+
+  let ge ik x y = not_bool @@ lt ik x y
+
+  let eq ik x y = match x, y with
+    | (a, b)::[], (c, d)::[] when (Ints_t.compare a b) == 0 && (Ints_t.compare c d) == 0 && (Ints_t.compare a c) == 0 -> 
+      one
+    | _ -> 
+      if is_bot (meet ik x y) then 
+        zero 
+      else 
+        top_bool
+
+  let ne ik x y = not_bool @@ eq ik x y
+  let interval_to_int i = Interval.to_int (Some i)
+  let interval_to_bool i = Interval.to_bool (Some i)
+
+  let log f ik (i1, i2) = 
+    match (interval_to_bool i1, interval_to_bool i2) with
+    | Some x, Some y -> of_bool ik (f x y)
+    | _ -> top_of ik
+
+  let log1 f ik i1 = 
+    match interval_to_bool i1 with
+    | Some x -> of_bool ik (f ik x)
+    | _ -> top_of ik
+
+  let bit f ik (i1, i2) =
+    match (interval_to_int i1), (interval_to_int i2) with
+    | Some x, Some y -> (try norm_interval ik (Interval.of_int ik (f ik x y)) with Division_by_zero -> top_of ik)
```

Comment:
Why supply `ik` to `f` if you then jump through hoops to make all `f` ignore the `ik`? Am I missing something?